[xGladiate] iP

xGladiate opened the pull request:
### XBot

> This is an extended version of Duke [What is Duke?](https://www.ibm.com/topics/chatbots)

- Create tasks using `todo`, `deadline`, `event` commands
- Label the tasks using `mark` or `unmark` commands
- Delete the tasks using `delete` command
- Show all tasks using `list` command

All you need is to 
1. Download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html)
2. Double-click it
3. Add your tasks
4. let it manage your tasks for you 😉

And it is _**FREE**_!!!
And also ~FAST~ _SUPER_ FAST to use!!!

**Features**: 
- [ ] Automatic storage of task
- [ ] Implement GUI

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java 
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

xGladiate committed:
Rename Duke to XBot

xGladiate committed:
Add greeting

xGladiate committed:
Add support for echo

xGladiate committed:
Fix bug in echo

xGladiate committed:
Add support for list, Modify location of input collection from scanner

xGladiate committed:
Add conditional statement to ensure nothing is added when input is empty

xGladiate committed:
Add Task class

xGladiate committed:
Move description format from XBot to Task

xGladiate committed:
Main: Modify conditional statement to switch

xGladiate committed:
Clean unnecessary code in Task

xGladiate committed:
Add error handling to invalid task number

xGladiate committed:
Add Deadline, Event and ToDo class

xGladiate committed:
Implement the ToDo, Event, Deadline class in XBot

xGladiate committed:
Add error handling for event & deadline invalid input format

xGladiate committed:
Modify display after subtype of Task is done / undone

xGladiate committed:
Add test case to automate testing & resolve bug

xGladiate committed:
Add XBotException.java

xGladiate committed:
Modify XBot to include XBotException

xGladiate committed:
Update Expected.txt after modification to exception

xGladiate committed:
Replace Arrays with Collections for list

xGladiate committed:
Add delete functionality

xGladiate committed:
Add testcases for delete

xGladiate committed:
Add TaskType (enum) to enhance readability for subtype of Task

xGladiate committed:
Add storage to store data of tasks

xGladiate committed:
Clean the code for the main function

xGladiate committed:
Add loadTask function and implement to the main

xGladiate committed:
Update enum to D, T, E instead

xGladiate committed:
Add access to the isDone variable in Task

xGladiate committed:
Add function to view the recorded tasks

xGladiate committed:
Fix the bug in deleteTask

xGladiate committed:
Modifiy the expected.txt for testing after bug fix for deleteTask

xGladiate committed:
Merge branch 'branch-Level-7'

xGladiate committed:
Modify print statement when there is no task

xGladiate committed:
Add changeDateFormat to Deadline.java

xGladiate committed:
Add error handling to dateInput for deadline

xGladiate committed:
Add more datetime formats to deadline

xGladiate committed:
Clean deadline to remove unnecessary code

xGladiate committed:
Add time change feature to event

xGladiate committed:
Add testcases for restriction to time input

xGladiate committed:
Merge branch 'branch-Level-8'

xGladiate committed:
Add Parser, Storage, TaskList, UI class and do basic categorisation

xGladiate committed:
Move isValidDateFormat to Parser

xGladiate committed:
Add packages into file and modify test settings

xGladiate committed:
Merge branch 'A-Packages'

xGladiate committed:
Merge branch 'add-gradle-support'

xGladiate committed:
Add JUnit tests

xGladiate committed:
Merge branch 'A-JUnit'

xGladiate committed:
Change xbot.Task to Task when print

xGladiate committed:
Add find function

xGladiate committed:
Merge branch 'branch-Level-9'

xGladiate committed:
Add javadoc comment to TaskList

xGladiate committed:
Add javadoc comment to XBot

xGladiate committed:
Add javadoc comment to XBotException

xGladiate committed:
Add javadoc comment to Parser

xGladiate committed:
Add javadoc commit to Storage

xGladiate committed:
Add javadoc commit to Deadline

xGladiate committed:
Add javadoc commit to Event

xGladiate committed:
Add javadoc commit to Task

xGladiate committed:
Add javadoc commit to TaskType

xGladiate committed:
Add javadoc commit to ToDo

xGladiate committed:
Add javadoc commit to Ui

xGladiate committed:
Move changeDateFormat to Parser

xGladiate committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/xbot/TaskList.java
#	src/main/java/xbot/ui/Ui.java

xGladiate committed:
Rearrange imports and deindent switch statements

xGladiate committed:
Rearrange imports inTaskList

xGladiate committed:
Rearrange imports in XBot

xGladiate committed:
Rearrange imports in Storage

xGladiate committed:
Rearrange imports in Deadline

xGladiate committed:
Rearrange imports in Event

xGladiate committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/xbot/TaskList.java
#	src/main/java/xbot/parser/Parser.java

zzzlou commented on src/main/java/xbot/TaskList.java:
Good job on not making any common mistakes! 💯

zzzlou commented on src/main/java/xbot/task/Task.java:
```diff
@@ -0,0 +1,89 @@
+    public boolean isDone() {
+        return isDone;
+    }
+    @Override
```
Probably should have an empty line before @Override?

zzzlou commented on src/main/java/xbot/task/Deadline.java:
good job!

zzzlou commented on src/main/java/xbot/XBot.java:
```diff
@@ -0,0 +1,48 @@
+
+        ui.showWelcome();
+        String input = ui.readCommand();
+        while(!input.equalsIgnoreCase("bye")) {
```
Should there be a space before ( ?

xGladiate committed:
Configure checkstyle

xGladiate committed:
Use Checkstyle with Gradle

xGladiate committed:
Fix checkstyle violations

kienvumrpm commented on src/main/java/xbot/parser/Parser.java:
```diff
@@ -0,0 +1,209 @@
+        formats.add("d/M/yyyy");
+        formats.add("d/M/yyyy HHmm");
+
+        for (String format : formats) {
```
I like the way you saved the timing variable (for Deadline and Event type) as String. It creates the flexibily to especially when only the date provided for as a deadline or start/end of an event.

xGladiate committed:
Add JavaFX Duke's interface

xGladiate committed:
Modify Duke to XBot

xGladiate committed:
Change CLI to GUI

xGladiate committed:
Merge branch 'branch-Level-10'

xGladiate committed:
Modify GUI interface dimension

Initially some of the elements do not automatically follow the dimension
of the app when resizing.

Modifying the anchor points of the elements is done such that the
elements of the app would be automatically resized when the dimensions
of the chatbot is modified.

xGladiate committed:
Customize the GUI interface

Changes were made to make the chatbot unique.

Added .css files: main.css and dialog-box.css,
and linked the .fxml files to the .css files to allow customization.

Changes made to allow customization:
* Change image cover
* Different color for background, text-input bar and buttons
* Speech-bubble added for conversation
* Modification of fonts to make the content more visible

xGladiate committed:
Add assertions

xGladiate committed:
Remove commented code

xGladiate committed:
Add UnknownTaskTypeException

xGladiate committed:
Refactor a new class for each command

xGladiate committed:
Refactor DeleteCommand to follow Code Quality

xGladiate committed:
Add streams to findTask method in TaskList

xGladiate committed:
Merge pull request #1 from xGladiate/branch-A-CodeQuality

Merge branch-A-CodeQuality into the master branch

xGladiate committed:
Merge pull request #3 from xGladiate/branch-A-Assertions

Merge branch-A-Assertions into the master branch

xGladiate committed:
Merge pull request #2 from xGladiate/branch-A-Streams

Merge branch-A-Streams into the master branch

xGladiate committed:
Set up Continuous Integration (CI)

xGladiate committed:
Merge branch 'branch-A-CI'

xGladiate committed:
Modify gradle.yml

xGladiate committed:
Debug load task from disk issue

xGladiate committed:
Add view command

xGladiate committed:
Modify view to include deadline tasks

xGladiate committed:
Merge branch 'branch-B-ViewSchedules'

xGladiate committed:
Remove invalid " " description

xGladiate committed:
Debug issues with error thrown for invalid taskNumber

xGladiate committed:
Change date output for view command

xGladiate committed:
Change reply to wrong time format

xGladiate committed:
Change output for marking task

xGladiate committed:
Add border color to button and text input

xGladiate committed:
Remove unnecessary test

xGladiate committed:
Change theme color

xGladiate committed:
Change avatar

xGladiate committed:
Merge branch 'branch-A-Personality'

xGladiate committed:
Customize the response to user input

xGladiate committed:
Merge branch 'branch-A-Personality'

xGladiate committed:
Merge branch 'branch-A-JUnit'

xGladiate committed:
Resolve bug for view

Resolved issues with regards to the allowed date and time format in
view command

xGladiate committed:
Modify README.md

xGladiate committed:
Add Ui.png into docs folder

xGladiate committed:
Modify code to adhere to checkstyle

xGladiate committed:
Modify JUnit to updated output

xGladiate committed:
Modify code to adhere to checkstyle

xGladiate committed:
Handle CI date-time parsing issues

xGladiate committed:
Modify gradle.yml to resolve date-time parsing issues

xGladiate committed:
peform gradle clean

xGladiate committed:
Modify DeadlineTest to resolve date-time parsing issues

xGladiate committed:
Resolve Checkstyle Violations

xGladiate committed:
Modify list command to remove header when there is no task

xGladiate committed:
Modify the name of jar file created

xGladiate committed:
Resolve format issues of empty task with valid view input

xGladiate committed:
Modify the icon and name of the application

xGladiate committed:
Remove Duke README.md

xGladiate committed:
Add greet message and bye message with closing of application

xGladiate committed:
Change the time before exiting application

xGladiate committed:
Add bye command to README.md

xGladiate committed:
Modify incorrect output for wrong format in event

xGladiate committed:
Resolve checkstyle violations

xGladiate committed:
Modify event and deadline command to throw error with empty description

xGladiate committed:
Modify Ui to the newer version and README to include sample ui

xGladiate committed:
Modify Ui to the newer version and README to include sample ui

xGladiate committed:
Add newline to the end of README file

xGladiate committed:
Merge branch 'branch-A-UserGuide' of https://github.com/xGladiate/ip

# Conflicts:
#	docs/README.md
